lib/rucio/transfertool/fts3.py

```diff
@@ -60,17 +61,62 @@ def __init__(self, external_host):
 
         :param external_host:   The external host where the transfertool API is running
         """
+        usercert = config_get('conveyor', 'usercert', False, None)
 
+        self.deterministic_id = config_get_bool('conveyor', 'use_deterministic_id', False, False)
         super(FTS3Transfertool, self).__init__(external_host)
         if self.external_host.startswith('https://'):
-            self.cert = (USERCERT, USERCERT)
+            self.cert = (usercert, usercert)
             self.verify = False
         else:
             self.cert = None
             self.verify = True  # True is the default setting of a requests.* method
 
     # Public methods part of the common interface
 
+    def delegate_proxy(self, proxy, ca_path='/etc/grid-security/certificates/', duration_hours=96, timeleft_hours=72):
+        """Delegate user proxy to fts server if the lifetime is less than timeleft_hours
+
+        :param proxy: proxy to be delegated
+        :type proxy: str
+
+        :param ca_path: ca path for verification, defaults to '/etc/grid-security/certificates/'
+        :param ca_path: str, optional
+        :param duration_hours: delegation validity duration in hours, defaults to 48
+        :param duration_hours: int, optional
+        :param timeleft_hours: minimal delegation time left, defaults to 12
+        :param timeleft_hours: int, optional
+
+        :return: delegation ID
+        :rtype: str
+        """
+        logging.info("Delegating proxy %s to %s", proxy, self.external_host)
+        start_time = time.time()
+
+        try:
+            context = Context(self.external_host,
+                              ucert=proxy,
+                              ukey=proxy,
+                              verify=True,
+                              capath=ca_path)
+            delegation_id = delegate(context,
+                                     lifetime=datetime.timedelta(hours=duration_hours),
+                                     delegate_when_lifetime_lt=datetime.timedelta(hours=timeleft_hours))
+            record_timer('transfertool.fts3.delegate_proxy.%s' % proxy, (time.time() - start_time))
```

Comment:
looks good!

can you maybe add extend it like this:

transfertool.fts3.delegate_proxy.success.%s

and below for the exceptions

transfertool.fts3.delegate_proxy.fail.%s

?

Reply:
sure, done